### PR TITLE
Banislav/knn regression

### PR DESCRIPTION
**Pull request by banislav**

### CODE REVIEW

File: knn-regression/knn-regression.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        self.k = k
+        self.n_features = n_features
+        self.noise = noise
+    
+    def load_data(self) -> None:
+        X, y = make_regression(n_features=self.n_features, noise=self.noise)
+        self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(X, y)
+        
+    def predict(self, x:List) -> float:
+        train_array = np.array(list(zip(self.X_train, self.y_train)), dtype=object)
```
Comment (daniildiveev) [inline knn-regression/knn-regression.py:20]: нахуя

File: knn-regression/knn-regression.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    def __init__(self, k:int=5, n_features:int=100, noise:int=50) -> None:
+        self.k = k
+        self.n_features = n_features
+        self.noise = noise
+    
+    def load_data(self) -> None:
+        X, y = make_regression(n_features=self.n_features, noise=self.noise)
+        self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(X, y)
+        
+    def predict(self, x:List) -> float:
```
Comment (daniildiveev) [inline knn-regression/knn-regression.py:19]: лист чего епта

сниппет :

```Python
def func(x:List[str/float/...])
```

File: knn-regression/knn-regression.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from sklearn.datasets import make_regression
+from sklearn.neighbors import KNeighborsRegressor
+from sklearn.model_selection import train_test_split
+from typing import List
```
Comment (daniildiveev) [inline knn-regression/knn-regression.py:7]: импорты по пеп 8

File: knn-regression/knn-regression.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        
+    def predict(self, x:List) -> float:
+        train_array = np.array(list(zip(self.X_train, self.y_train)), dtype=object)
+        predict_values = np.array([])
+        
+        if x.shape[1] != self.X_train.shape[1]:
+            raise ValueError("value x is not suitable")
+            
+        for item in x:
+            norm_list = [(np.linalg.norm(item - train_array[i][0]), train_array[i][1]) for i in range(train_array.shape[0])]
```
Comment (daniildiveev) [inline knn-regression/knn-regression.py:27]: `np.linalg.norm` можно без цикла делать через `axis` 

```python
np.linalg.norm(self.x - x_test[i], axis=1) ...
```

File: knn-regression/knn-regression.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            norm_list = sorted(norm_list, key=lambda x: x[0])
```
Comment (daniildiveev) [inline knn-regression/knn-regression.py:28]: здесь надо бы `argsort` делать,  так проще и потом просто передавать превые к индексов в квадратные скобки

сниппет:

```python
k = 2
a = np.array([ 4, 6, 0, 9])
sorted_args = a.argsort()  # [2, 0, 1, 3]
nearest_neigbors = sorted_args[:k] # [2, 0]
a[nearest_neigbors]  # [0, 4]
```

File: knn-regression/knn-regression.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import pandas as pd
+from sklearn.datasets import make_regression
+from sklearn.neighbors import KNeighborsRegressor
+from sklearn.model_selection import train_test_split
+from typing import List
+
+class KnnRegressor:
+    def __init__(self, k:int=5, n_features:int=100, noise:int=50) -> None:
+        self.k = k
+        self.n_features = n_features
```
Comment (daniildiveev) [inline knn-regression/knn-regression.py:12]: я бы noise  и n_features всунул в метод лоад дата, потому что если я захочу новые данные мне придется инстанс пересоздавать

File: knn-regression/knn-regression.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        self.noise = noise
+    
+    def load_data(self) -> None:
+        X, y = make_regression(n_features=self.n_features, noise=self.noise)
+        self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(X, y)
```
Comment (daniildiveev) [inline knn-regression/knn-regression.py:17]: train_test_split очевидно нужно делать не внутри класса

**Review by daniildiveev — CHANGES_REQUESTED** [review]
